workers/src/executor_def.rs

```diff
@@ -0,0 +1,152 @@
+
+        loop {
+            tokio::select! {
+                Some(log) = ps_stream.next() => {
+                    match log.log_decode::<PacketSent>() {
+                        Ok(packet_log) => {
+                            self.packet_queue.push_back(packet_log.data().clone());
+                        },
+                        Err(e) => { error!("Failed to decode PacketSent event: {:?}", e);}
+                    }
+                }
+                Some(log) = ef_stream.next() => {
+                    match log.log_decode::<ExecutorFeePaid>() {
+                        Ok(executor_fee_log) => {
+                            if self.packet_queue.is_empty() {
+                                continue;
+                            }
+
+                            if !executor_fee_log.data().executor.eq(&self.config.dvn_addr)  {
+                                self.packet_queue.clear();
+                                continue;
+                            }
+                        },
+                        Err(e) => { error!("Failed to decode ExecutorFeePaid event: {:?}", e);}
+                    }
+                },
+                Some(log) = pv_stream.next() => {
+                    match log.log_decode::<PacketVerified>() {
+                        Ok(inner_log) => {
+                           Self::handle_verified_packet(&contract, &mut self.packet_queue, inner_log.data()).await?;
+                        },
+                        Err(e) => { error!("Failed to decode PacketVerified event: {:?}", e);}
+                    }
+                },
+            }
+            if self.finish {
+                break;
+            }
+        }
+        Ok(())
+    }
+
+    async fn handle_verified_packet(
+        contract: &ContractInst,
+        queue: &mut VecDeque<PacketSent>,
+        packet_verified: &PacketVerified,
+    ) -> Result<()> {
+        if queue.is_empty() {
+            return Ok(());
+        }
+
+        let packet_sent = queue.pop_front().unwrap();
+        // We don't expect any more items to be present. If we have any - they are garbage/leftovers.
+        queue.clear();
+        // Despite being described with other arguments, the only real implementation of
+        //  `executable` function is in the contract located here: https://shorturl.at/4H6Yz
+        // function executable(Origin memory _origin, address _receiver) returns (ExecutionState)
+        let call_builder = contract.function(
+            "executable",
+            &[
+                prepare_header(&packet_sent.encodedPayload[..]),
+                DynSolValue::Address(packet_verified.receiver),
+            ],
+        )?;
+
+        let mut retry_count = 0;
+        // status `Executable` is represented by the integer 2 in the enum.
+        // To read more: https://tinyurl.com/zur3btzs (line 9)
+        let executable: &DynSolValue = &DynSolValue::Int(I256::unchecked_from(2), 32);
```

Comment:
Should we make this into a constant?

Reply:
I tried, but impossible. Can't be a constant by invariant, unfortunately.